.github/scripts/build-macos.sh

```diff
@@ -9,4 +15,6 @@ cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo -DBUILD_TESTS=ON -DDETHRACE_WERROR=ON -B
 cmake --build build -- -j 4
```

Comment:
I think you need to add `-DCMAKE_OSX_ARCHITECTURES=x86_64` (or `amd64` idk) to be sure this does not build an armv8 executable.
Speaking of armv8, perhaps add `armv8` to the build table for macos?

See [`CMAKE_OSX_ARCHITECTURES`](https://cmake.org/cmake/help/latest/variable/CMAKE_OSX_ARCHITECTURES.html).
I think you can build a "fat" executable by doing `-DCMAKE_OSX_ARCHITECTURES="x86_64;armv8"`, but don't quote me on that :)

Reply:
thanks. github actions don't yet support arm osx builds (https://github.com/actions/runner-images/issues/2187), so I've set it just to `x86_x64`

Reply:
We don't need M1 hardware. We just want to cross build to armv8.

Reply:
Actually, we do when we run the tests. So you're right.